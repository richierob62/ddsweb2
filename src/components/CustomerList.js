import React from 'react';
import Table from 'react-bootstrap/lib/Table';
import { connect } from 'react-redux';
import * as sel from '../selectors';
import * as act from '../actions';

const cl = (props) => {

    const isSelected = (id) => props.selected_id === id ? 'selected' : '';

    const sortIndicator = (column_name) => {
        if(props.sort_field !== column_name ) return '';
        return props.sort_direction === 'ASC' ? 
        <i className="fa fa-sort-asc" aria-hidden="true"></i> : 
        <i className="fa fa-sort-desc" aria-hidden="true"></i> ;
    };
    const renderRows = () => {
        return props.list.map(row => {
            return (
                <tr id={'data-row-' + row.get('id') }
                    key={row.get('id')}
                    className={isSelected(row.get('id'))}
                    onClick={props.setSelected.bind(null, row.get('id'))}>
                    <td id={'table-body-customer-col-1'}>{row.get('name')}</td>
                    <td id={'table-body-customer-col-2'}>{row.get('address')}</td>
                    <td id={'table-body-customer-col-3'}>{row.get('city')}</td>
                    <td id={'table-body-customer-col-4'}>{row.get('state')}</td>
                    <td id={'table-body-customer-col-5'}>{row.get('zip')}</td>
                    <td id={'table-body-customer-col-6'}>{row.get('area')}</td>
                    <td id={'table-body-customer-col-7'}>{row.get('phone')}</td>
                    <td id={'table-body-customer-col-8'}>{row.get('account_num')}</td>
                    <td id={'table-body-customer-col-9'}>{row.get('sales_rep')}</td>
                </tr>
            );
        });
    };

    const filterChange = (column, e) => props.changeFilter(column, e.currentTarget.value);

    const setFilterValue = (column) => props.current_filters.get(column) ? props.current_filters.get(column) : '';

    return (
        <Table striped bordered condensed hover className={'table-fixedheader'}>
            <thead>
                <tr>
                    <th id={'table-filter-customer-col-1'} ><input id={'rass'} type='text' value={setFilterValue('name')} onChange={filterChange.bind(null, 'name')} /></th>
                    <th id={'table-filter-customer-col-2'} ><input type='text' value={setFilterValue('address')} onChange={filterChange.bind(null, 'address')} /></th>
                    <th id={'table-filter-customer-col-3'} ><input type='text' value={setFilterValue('city')} onChange={filterChange.bind(null, 'city')} /></th>
                    <th id={'table-filter-customer-col-4'} ><input type='text' value={setFilterValue('state')} onChange={filterChange.bind(null, 'state')} /></th>
                    <th id={'table-filter-customer-col-5'} ><input type='text' value={setFilterValue('zip')} onChange={filterChange.bind(null, 'zip')} /></th>
                    <th id={'table-filter-customer-col-6'} ><input type='text' value={setFilterValue('area')} onChange={filterChange.bind(null, 'area')} /></th>
                    <th id={'table-filter-customer-col-7'} ><input type='text' value={setFilterValue('phone')} onChange={filterChange.bind(null, 'phone')} /></th>
                    <th id={'table-filter-customer-col-8'} ><input type='text' value={setFilterValue('account_num')} onChange={filterChange.bind(null, 'account_num')} /></th>
                    <th id={'table-filter-customer-col-9'} ><input type='text' value={setFilterValue('sales_rep')} onChange={filterChange.bind(null, 'sales_rep')} /></th>
                </tr>
                <tr>
                    <th id={'table-head-customer-col-1'} onClick={props.changeSort.bind(null, 'name')} >Company Name {sortIndicator('name')} </th>
                    <th id={'table-head-customer-col-2'} onClick={props.changeSort.bind(null, 'address')} >Address {sortIndicator('address')} </th>
                    <th id={'table-head-customer-col-3'} onClick={props.changeSort.bind(null, 'city')} >City {sortIndicator('city')} </th>
                    <th id={'table-head-customer-col-4'} onClick={props.changeSort.bind(null, 'state')} >State {sortIndicator('state')} </th>
                    <th id={'table-head-customer-col-5'} onClick={props.changeSort.bind(null, 'zip')} >Zip {sortIndicator('zip')} </th>
                    <th id={'table-head-customer-col-6'} onClick={props.changeSort.bind(null, 'area')} >Area {sortIndicator('area')} </th>
                    <th id={'table-head-customer-col-7'} onClick={props.changeSort.bind(null, 'phone')} >Phone {sortIndicator('phone')} </th>
                    <th id={'table-head-customer-col-8'} onClick={props.changeSort.bind(null, 'account_num')} >Acc # {sortIndicator('account_num')} </th>
                    <th id={'table-head-customer-col-9'} onClick={props.changeSort.bind(null, 'sales_rep')} >Sales Rep {sortIndicator('sales_rep')} </th>
                </tr>
            </thead>
            <tbody>
                {renderRows()}
            </tbody>
        </Table>
    );
};

const mapStateToProps = (state) => {
    return {
        list: sel.getCustomerList(state),
        selected_id: sel.getSelectedCustomerID(state),
        sort_field: sel.getCustomerSortField(state),
        sort_direction: sel.getCustomerSortDirection(state),
        current_filters: sel.getCustomerFilters(state),
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setSelected(id) { dispatch(act.selectCustomer(id)); },
        changeSort(column) { dispatch(act.changeCustomerSort(column)); },
        changeFilter(column, value) { dispatch(act.changeCustomerFilter(column, value)); }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(cl);