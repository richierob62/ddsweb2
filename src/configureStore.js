import { createStore, applyMiddleware, compose } from 'redux';
// import createSagaMiddleware, { END } from 'redux-saga';
import rootReducer from './reducers/rootReducer';

const configureStore = () => {

    //   const sagaMiddleware = createSagaMiddleware();

    //   const middlewareList = global.running_tests
    //     ? applyMiddleware(sagaMiddleware)
    //     : applyMiddleware(sagaMiddleware, createLogger());

    //   const store = createStore(
    //     rootReducer,
    //     compose(
    //       middlewareList,
    //       window.devToolsExtension ? window.devToolsExtension() : f => f
    //     )
    //   );

    //   if (module.hot) {
    //     // Enable Webpack hot module replacement for reducer
    //     module.hot.accept('./reducer', () => {
    //       const nextRootReducer = require('./reducer').default;
    //       store.replaceReducer(nextRootReducer);
    //     });
    //   }

    //   store.runSaga = sagaMiddleware.run;
    //   store.close = () => store.dispatch(END);
    //   return store;

    return createStore(rootReducer,  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

};

export default configureStore;