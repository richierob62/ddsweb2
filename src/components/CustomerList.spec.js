import React from 'react';
import CustomerList from './CustomerList';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from '../configureStore';

const store = configureStore();

it('renders without crashing', () => {
  const el = shallow(<CustomerList store={store}/>);
  expect(shallowToJson(el)).toMatchSnapshot();
});

