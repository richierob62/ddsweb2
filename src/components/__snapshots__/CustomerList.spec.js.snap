exports[`test renders without crashing 1`] = `
<customerTable
  changeFilter={[Function]}
  changeSort={[Function]}
  current_filters={Object {}}
  list={
    Array [
      Object {
        "account_num": "4041234",
        "address": "10 Leeton Ridge Ave. ",
        "area": "494",
        "city": "Wisconsin Rapids",
        "id": 1,
        "name": "L\'Oréal",
        "phone": "487-8119",
        "sales_rep": 3,
        "state": "WI",
        "zip": "54494",
      },
      Object {
        "account_num": "4130144",
        "address": "121 NW. Indian Spring Street ",
        "area": "664",
        "city": "Lebanon",
        "id": 2,
        "name": "Adobe Systems",
        "phone": "810-1662",
        "sales_rep": 7,
        "state": "PA",
        "zip": "17042",
      },
      Object {
        "account_num": "9842555",
        "address": "164 Depot Ave. ",
        "area": "440",
        "city": "North Ridgeville",
        "id": 3,
        "name": "Allianz",
        "phone": "215-4916",
        "sales_rep": 9,
        "state": "OH",
        "zip": "44039",
      },
      Object {
        "account_num": "1028587",
        "address": "2 Kirkland St. ",
        "area": "601",
        "city": "Matawan",
        "id": 4,
        "name": "Amazon.com",
        "phone": "777-4414",
        "sales_rep": 4,
        "state": "NJ",
        "zip": "7747",
      },
      Object {
        "account_num": "8652533",
        "address": "204 East Ryan St. ",
        "area": "517",
        "city": "Braintree",
        "id": 5,
        "name": "American Express",
        "phone": "185-6539",
        "sales_rep": 10,
        "state": "MA",
        "zip": "2184",
      },
      Object {
        "account_num": "4816950",
        "address": "251 Shub Farm Drive ",
        "area": "419",
        "city": "Huntington Station",
        "id": 6,
        "name": "AT&T",
        "phone": "616-9065",
        "sales_rep": 9,
        "state": "NY",
        "zip": "11746",
      },
      Object {
        "account_num": "5828729",
        "address": "28 River St. ",
        "area": "144",
        "city": "Arlington Heights",
        "id": 7,
        "name": "Bank of America",
        "phone": "878-8165",
        "sales_rep": 8,
        "state": "IL",
        "zip": "60004",
      },
      Object {
        "account_num": "4467658",
        "address": "29 Grant St. ",
        "area": "601",
        "city": "Hicksville",
        "id": 8,
        "name": "Budweiser Stag Brewing Company",
        "phone": "750-8542",
        "sales_rep": 3,
        "state": "NY",
        "zip": "11801",
      },
      Object {
        "account_num": "1981765",
        "address": "3 Country Court ",
        "area": "340",
        "city": "Bismarck",
        "id": 9,
        "name": "Caterpillar Inc.",
        "phone": "671-3497",
        "sales_rep": 8,
        "state": "ND",
        "zip": "58501",
      },
      Object {
        "account_num": "6702723",
        "address": "30 Brook Ave. ",
        "area": "587",
        "city": "Waterford",
        "id": 10,
        "name": "Chase",
        "phone": "596-6468",
        "sales_rep": 8,
        "state": "MI",
        "zip": "48329",
      },
      Object {
        "account_num": "8145803",
        "address": "313 Paris Hill Road ",
        "area": "386",
        "city": "Skokie",
        "id": 11,
        "name": "Cisco Systems, Inc.",
        "phone": "962-4376",
        "sales_rep": 5,
        "state": "IL",
        "zip": "60076",
      },
      Object {
        "account_num": "9347709",
        "address": "353 Washington Ave. ",
        "area": "452",
        "city": "Trenton",
        "id": 12,
        "name": "Citigroup",
        "phone": "543-4590",
        "sales_rep": 10,
        "state": "NJ",
        "zip": "8610",
      },
      Object {
        "account_num": "8712262",
        "address": "355 Andover Court ",
        "area": "175",
        "city": "Saint Paul",
        "id": 13,
        "name": "Coca-Cola",
        "phone": "150-0787",
        "sales_rep": 4,
        "state": "MN",
        "zip": "55104",
      },
      Object {
        "account_num": "5465816",
        "address": "4 Buckingham Road ",
        "area": "157",
        "city": "Vicksburg",
        "id": 14,
        "name": "Corona",
        "phone": "380-6555",
        "sales_rep": 5,
        "state": "MS",
        "zip": "39180",
      },
      Object {
        "account_num": "5252587",
        "address": "40 Glenridge Street ",
        "area": "664",
        "city": "North Tonawanda",
        "id": 15,
        "name": "Credit Suisse",
        "phone": "519-7267",
        "sales_rep": 7,
        "state": "NY",
        "zip": "14120",
      },
      Object {
        "account_num": "2504981",
        "address": "429 Fairfield St. ",
        "area": "879",
        "city": "Urbandale",
        "id": 16,
        "name": "Deere & Company",
        "phone": "131-0370",
        "sales_rep": 9,
        "state": "IA",
        "zip": "50322",
      },
      Object {
        "account_num": "2988422",
        "address": "43 Thorne Dr. ",
        "area": "244",
        "city": "Tampa",
        "id": 17,
        "name": "eBay",
        "phone": "569-9699",
        "sales_rep": 7,
        "state": "FL",
        "zip": "33604",
      },
      Object {
        "account_num": "1288369",
        "address": "45 Essex Drive ",
        "area": "826",
        "city": "Collegeville",
        "id": 18,
        "name": "Global Gillette",
        "phone": "860-7829",
        "sales_rep": 9,
        "state": "PA",
        "zip": "19426",
      },
      Object {
        "account_num": "5040201",
        "address": "471 Lake Forest St. ",
        "area": "577",
        "city": "Warwick",
        "id": 19,
        "name": "Harley-Davidson Motor Company",
        "phone": "835-2499",
        "sales_rep": 7,
        "state": "RI",
        "zip": "2886",
      },
      Object {
        "account_num": "9216443",
        "address": "51 Oakwood Lane ",
        "area": "704",
        "city": "Kansas City",
        "id": 20,
        "name": "Hermès",
        "phone": "170-7774",
        "sales_rep": 6,
        "state": "MO",
        "zip": "64151",
      },
      Object {
        "account_num": "1427848",
        "address": "520 Wayne Street ",
        "area": "527",
        "city": "Buford",
        "id": 21,
        "name": "Hewlett-Packard",
        "phone": "940-8976",
        "sales_rep": 1,
        "state": "GA",
        "zip": "30518",
      },
      Object {
        "account_num": "5679667",
        "address": "544 Hillcrest Street ",
        "area": "811",
        "city": "Palos Verdes Peninsula",
        "id": 22,
        "name": "Home Depot",
        "phone": "718-4893",
        "sales_rep": 5,
        "state": "CA",
        "zip": "90274",
      },
      Object {
        "account_num": "2006699",
        "address": "6 S. Ohio Street ",
        "area": "377",
        "city": "Trumbull",
        "id": 23,
        "name": "IBM",
        "phone": "749-5649",
        "sales_rep": 8,
        "state": "CT",
        "zip": "6611",
      },
      Object {
        "account_num": "5835848",
        "address": "615 South Newbridge Drive ",
        "area": "319",
        "city": "West Bloomfield",
        "id": 24,
        "name": "Intel Corporation",
        "phone": "834-5315",
        "sales_rep": 10,
        "state": "MI",
        "zip": "48322",
      },
      Object {
        "account_num": "5677722",
        "address": "620 Golden Star St. ",
        "area": "770",
        "city": "Mundelein",
        "id": 25,
        "name": "Jack Daniel\'s",
        "phone": "869-2784",
        "sales_rep": 3,
        "state": "IL",
        "zip": "60060",
      },
      Object {
        "account_num": "9823346",
        "address": "72 Cross St. ",
        "area": "194",
        "city": "Newburgh",
        "id": 26,
        "name": "Johnnie Walker",
        "phone": "345-7191",
        "sales_rep": 10,
        "state": "NY",
        "zip": "12550",
      },
      Object {
        "account_num": "5679108",
        "address": "7310 Orange Drive ",
        "area": "705",
        "city": "Oshkosh",
        "id": 27,
        "name": "KFC",
        "phone": "497-0049",
        "sales_rep": 1,
        "state": "WI",
        "zip": "54901",
      },
      Object {
        "account_num": "7992192",
        "address": "7399 Bridgeton Ave. ",
        "area": "648",
        "city": "Danbury",
        "id": 28,
        "name": "MasterCard",
        "phone": "617-4166",
        "sales_rep": 1,
        "state": "CT",
        "zip": "6810",
      },
      Object {
        "account_num": "3380707",
        "address": "74 Clinton Ave. ",
        "area": "826",
        "city": "Owensboro",
        "id": 29,
        "name": "McDonald\'s",
        "phone": "928-7002",
        "sales_rep": 8,
        "state": "KY",
        "zip": "42301",
      },
      Object {
        "account_num": "5339897",
        "address": "74 Ramblewood Ave. ",
        "area": "270",
        "city": "Lanham",
        "id": 30,
        "name": "Mercedes-Benz",
        "phone": "934-0454",
        "sales_rep": 2,
        "state": "MD",
        "zip": "20706",
      },
      Object {
        "account_num": "5260880",
        "address": "7480 Santa Clara St. ",
        "area": "261",
        "city": "North Kingstown",
        "id": 31,
        "name": "Microsoft",
        "phone": "199-2631",
        "sales_rep": 8,
        "state": "RI",
        "zip": "2852",
      },
      Object {
        "account_num": "7358127",
        "address": "7517 High Noon Street ",
        "area": "409",
        "city": "Danvers",
        "id": 32,
        "name": "Moët et Chandon",
        "phone": "428-6706",
        "sales_rep": 5,
        "state": "MA",
        "zip": "1923",
      },
      Object {
        "account_num": "5358787",
        "address": "759 Tarkiln Hill Street ",
        "area": "425",
        "city": "San Lorenzo",
        "id": 33,
        "name": "Nintendo",
        "phone": "895-0818",
        "sales_rep": 5,
        "state": "CA",
        "zip": "94580",
      },
      Object {
        "account_num": "8372416",
        "address": "77 Saxon Drive ",
        "area": "695",
        "city": "Floral Park",
        "id": 34,
        "name": "Nissan Motor Co., Ltd.",
        "phone": "754-6120",
        "sales_rep": 1,
        "state": "NY",
        "zip": "11001",
      },
      Object {
        "account_num": "7893213",
        "address": "7748 Buckingham Street ",
        "area": "971",
        "city": "Fremont",
        "id": 35,
        "name": "Nokia",
        "phone": "680-1133",
        "sales_rep": 3,
        "state": "OH",
        "zip": "43420",
      },
      Object {
        "account_num": "2145343",
        "address": "787 Hawthorne St. ",
        "area": "600",
        "city": "Danville",
        "id": 36,
        "name": "NTT Data",
        "phone": "321-3720",
        "sales_rep": 10,
        "state": "VA",
        "zip": "24540",
      },
      Object {
        "account_num": "6607691",
        "address": "7898 Redwood Drive ",
        "area": "663",
        "city": "Columbia",
        "id": 37,
        "name": "Pampers",
        "phone": "389-6517",
        "sales_rep": 1,
        "state": "MD",
        "zip": "21044",
      },
      Object {
        "account_num": "3074615",
        "address": "7940 Vale Avenue ",
        "area": "788",
        "city": "York",
        "id": 38,
        "name": "Porsche",
        "phone": "130-8139",
        "sales_rep": 9,
        "state": "PA",
        "zip": "17402",
      },
      Object {
        "account_num": "2508625",
        "address": "8 Fawn Avenue ",
        "area": "471",
        "city": "Londonderry",
        "id": 39,
        "name": "Prada",
        "phone": "265-6066",
        "sales_rep": 10,
        "state": "NH",
        "zip": "3053",
      },
      Object {
        "account_num": "7812640",
        "address": "80 Bayport Lane ",
        "area": "149",
        "city": "South El Monte",
        "id": 40,
        "name": "Ralph Lauren Corporation",
        "phone": "261-2423",
        "sales_rep": 10,
        "state": "CA",
        "zip": "91733",
      },
      Object {
        "account_num": "3239194",
        "address": "8303 Buckingham Street ",
        "area": "756",
        "city": "Bozeman",
        "id": 41,
        "name": "Siemens AG",
        "phone": "812-9276",
        "sales_rep": 3,
        "state": "MT",
        "zip": "59715",
      },
      Object {
        "account_num": "4347061",
        "address": "861 Jones Rd. ",
        "area": "422",
        "city": "New Brunswick",
        "id": 42,
        "name": "Smirnoff",
        "phone": "154-0927",
        "sales_rep": 1,
        "state": "NJ",
        "zip": "8901",
      },
      Object {
        "account_num": "5396336",
        "address": "8936 Joy Ridge Circle ",
        "area": "301",
        "city": "Glendale Heights",
        "id": 43,
        "name": "Sony",
        "phone": "702-9390",
        "sales_rep": 3,
        "state": "IL",
        "zip": "60139",
      },
      Object {
        "account_num": "1615106",
        "address": "9 N. Wayne Drive ",
        "area": "417",
        "city": "Sandusky",
        "id": 44,
        "name": "Sprite",
        "phone": "582-9423",
        "sales_rep": 1,
        "state": "OH",
        "zip": "44870",
      },
      Object {
        "account_num": "6714269",
        "address": "916 Mayflower St. ",
        "area": "972",
        "city": "Vienna",
        "id": 45,
        "name": "Tiffany & Co.",
        "phone": "489-1680",
        "sales_rep": 7,
        "state": "VA",
        "zip": "22180",
      },
      Object {
        "account_num": "9299734",
        "address": "9377 Catherine Ave. ",
        "area": "906",
        "city": "Elizabethtown",
        "id": 46,
        "name": "United Parcel Service",
        "phone": "702-7893",
        "sales_rep": 3,
        "state": "PA",
        "zip": "17022",
      },
      Object {
        "account_num": "8559141",
        "address": "9402 W. Ohio St. ",
        "area": "800",
        "city": "Colonial Heights",
        "id": 47,
        "name": "Verizon Communications",
        "phone": "356-1100",
        "sales_rep": 4,
        "state": "VA",
        "zip": "23834",
      },
      Object {
        "account_num": "2512935",
        "address": "9429 Arnold Drive ",
        "area": "758",
        "city": "Highland Park",
        "id": 48,
        "name": "Vodafone",
        "phone": "747-0954",
        "sales_rep": 6,
        "state": "IL",
        "zip": "60035",
      },
      Object {
        "account_num": "1500730",
        "address": "9519 Arnold St. ",
        "area": "463",
        "city": "Highland Park",
        "id": 49,
        "name": "Wal-Mart",
        "phone": "462-7023",
        "sales_rep": 3,
        "state": "IL",
        "zip": "60035",
      },
      Object {
        "account_num": "9283560",
        "address": "970 Pine St. ",
        "area": "139",
        "city": "Centreville",
        "id": 50,
        "name": "Yahoo!",
        "phone": "865-2178",
        "sales_rep": 1,
        "state": "VA",
        "zip": "20120",
      },
    ]
  }
  selected_id={-1}
  setSelected={[Function]}
  sort_direction="ASC"
  sort_field="name"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
